src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private HashObject<K, V>[] hashMap;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size = 0;
+
+    public MyHashMap() {
+        hashMap = new HashObject[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    public K[] keySet() {
+        if (size == 0) {
+            return null;
+        } else {
+            K[] keys = (K[]) new Object[size];
+            int index = 0;
+            for (HashObject<K, V> object : hashMap) {
+                if (object != null) {
+                    HashObject<K,V> currnetNode = object;
```

Comment:
The name `currnetNode` is a typo, it should be `currentNode`. Remember to use informative variable names and check for typos.